pandas/formats/style.py

```diff
@@ -10,7 +10,9 @@
 from collections import defaultdict, MutableMapping
 
 try:
-    from jinja2 import Template
+    from jinja2 import (
+        PackageLoader, Environment, ChoiceLoader, FileSystemLoader
+    )
 except ImportError:
     msg = "pandas.Styler requires jinja2. "\
```

Comment:
@jreback didn't quite work since you can't early-return from a module.

However, this does work:

```python
try:
    from pandas.formats.style import Styler
except:
    from pandas.compat import add_metaclass

    # We want to *not* raise an ImportError upon init
    # We *do* want to raise an ImportError with a custom message
    # when the class is instantiated or subclassed.

    _msg = ("pandas.io.api.Styler requires jinja2. "
            "Please install with `conda install Jinja2` "
            "or `pip install Jinaj2`")

    class _Check(type):
        """Raises import error upon subclassing."""
        def __init__(cls, name, bases, clsdict):
            if len(cls.mro()) > 2:
                raise ImportError(_msg)
            super(_Check, cls).__init__(name, bases, clsdict)

    @add_metaclass(_Check)
    class Styler(object):
        def __init__(self, *args, **kargs):
            raise ImportError(_msg)

```

Are we ok with a metaclass, just for a nice error message? :D